other(editor): Integrated the modification of identifiers and aliases into the primary editing workflow

Tarunmeena0901 opened the pull request:
<!--
Before making a pull request, please:
1. Read the guidelines for contributing: https://github.com/metabrainz/guidelines/blob/master/GitHub.md
2. Verify that your changes match our coding style
3. Fill out the requested information
-->

### Problem
<!-- What are you trying to solve?
Add the JIRA ticket number if you are working on one --> [BB-343]


### Solution
<!-- What does this PR do to fix the problem? -->
Enhanced main entity editor page:
- Introduced an identifier editing section.
- Added a section for adding aliases, improving overall editing flow.
- Aligned the remove button with identifier rows for a more polished appearance.

### Areas of Impact
<!-- What parts of the codebase and which behaviors are affected? -->
~ identifier-editor.js
~ identifier-row.tsx
~ alias-editor.js
~ entity-editor.tsx

![editor-page](https://github.com/metabrainz/bookbrainz-site/assets/97682967/694f6040-4783-4161-af23-433d6be5fcb0)

Tarunmeena0901 commented:
i fixed the "add new identifier" in alias editing section as u can see in screenshots

MonkeyDo commented on src/client/entity-editor/alias-editor/alias-editor.js:
```diff
@@ -37,15 +35,14 @@ import {faQuestionCircle} from '@fortawesome/free-solid-svg-icons';
 	languageOptions,
-	onClose,
-	show
+	onClose
```
The name onClose needs to be changed everywhere since the function has changed.
What does it do now?

Tarunmeena0901 replied:
ok so this onClose is a dispatch function which is used to remove any empty alias row when user click on Done button (refer SS), 
I thought to change the name to onDone but it doesn't seem very descriptive to me so I thought it may be good to hear some review

MonkeyDo commented on src/client/entity-editor/identifier-editor/identifier-editor.js:
```diff
@@ -67,37 +63,34 @@ const IdentifierEditor = ({
+		<div>
+			<div>
+				<div style={{display: 'flex'}}>
+					<h2>Add new indentifiers</h2>
```
```suggestion
					<h2>Add identifiers</h2>
```

MonkeyDo commented on src/client/entity-editor/identifier-editor/identifier-editor.js:
```diff
@@ -67,37 +63,34 @@ const IdentifierEditor = ({
-};
-IdentifierEditor.defaultProps = {
-	show: false
+	onClose: PropTypes.func.isRequired
```
Same comment as above, what does this do now?

MonkeyDo reviewed (changes requested):
Thanks for opening a PR @Tarunmeena0901 !
Sorry it took so long to get to reviewing it.

Currently, my main feedback looking at the screenshot you posted is:
- each row is taking too much space now, both vertically and horizontally; it made some sense in the modals but in the page we need to conserve some space. For example the giant remove button is unnecessary and can be a small trash icon instead; there is too much padding in between rows; the inputs are quite large, etc.
Maybe the sections do not need to take the whole width of the page; I think both would do better with half of the page width (on desktop).
- the alias section I think would make more sense next to the name section; after all they are alternate names, it seems odd to me to move that section to the bottom of the page. Looking at the name section component, you might need to refactor the layout if you want to move things around or put the two sections side by side.

Tarunmeena0901 commented:
I understand everyone is very busy after vacation so please take your time for reviewing my PR's 😄 
I do wanted to propose a design for this issue but I was waiting for a reply as I didn't wanted to tag any one , 
I'll be back with a design proposal

MonkeyDo commented:
Sorry for the delay on reviewing this; I think it deserves a bit more work in terms of design, I think the identifiers section as laid out currently takes a lot of space (I think this was most likely the reason why it was put in a modal before).
It might require a more general reworking of the edit page, and as you mentioned it also needs to take the book wizard page into account.

Are you up for the challenge of proposing a mockup for improving the page design so that this new section fits better?
If not I am happy to work on it but it will most likely take me a couple of weeks to get to.

Tarunmeena0901 commented:
> It might require a more general reworking of the edit page, and as you mentioned it also needs to take the book wizard page into account.

I guess I'll have to sit down with my pencil paper and think this out properly. This week is pretty loaded, but next week won't be. I'll ping you soon...

Tarunmeena0901 commented:
apologies for taking so long for implementing the mockups , there was more work then i expected . the changes in git may seems confusing so i am mentioning some main changes i made
- i removed the button bar component from entity editor page as it was not needed any more but i still deleted it completely the folder and files still exist
- changes may show some files are deleted and added but that's only because of renaming of parent folder, for ex
alias-editor => alias-section
- I added a new redux action and case to add new alias and and new identifiers and removed the addAliasRow/addIdentifierRow from redux
- added a new modal to edit the existing identifiers and aliases
- renamed identifierEditor and aliasEditor variable in backend and test files to identifierSection and aliasSection

![Screenshot 2024-04-24 035858](https://github.com/metabrainz/bookbrainz-site/assets/97682967/16edad9b-7726-451d-8da2-6abbb0db0431)

![Screenshot 2024-04-24 035843](https://github.com/metabrainz/bookbrainz-site/assets/97682967/aaf9faa7-0aa7-4f51-9d70-e654079fb2b0)

![Screenshot 2024-04-23 230944](https://github.com/metabrainz/bookbrainz-site/assets/97682967/70b5f5cb-6276-41e0-a8d8-5767444ab783)

MonkeyDo commented:
Hello, thanks again for working on this, it's not an easy issue!
I have tested the PR locally, and I think there is some room for improvement in terms of UI and UX.
That being said, I did send you off with basically no direction on that subject, so I blame myself. The coding aspects look fine.

I need to sit on it for a bit and do some experimentation to come up with a more detailed mockup of what the pages should look like.
I reckon some of that experimentation I will have to do directly in the code to make it easier, in which case I could possibly directly push my changes if that's OK with you.
Otherwise, If you would prefer to do the changes yourself, I can do my best to give you better mockups and some clear directions for the code if you prefer.
What do you think?

Tarunmeena0901 commented:
If you have specific mockups or ideas in mind, I am more than happy to implement them. However, if you find it easier to experiment directly with the code, please feel free to push your changes to the PR.

Given that the GSoC coding period is ongoing and you might have a lot on your plate, please let me know how you'd prefer to proceed. I'm open to either option.
